docs/9_0/components/panelgrid.md

```diff
@@ -103,6 +103,56 @@ However, you may find your labels not lining up in horizontal forms.  You can us
              styleClass="customPanelGrid">
 ```
 
+## Responsive Style, StyleClass, and ID
+PanelGrid offers responsive support using `layout="flex"` for PrimeFlex or `layout="grid"` for Grid CSS and
+ a non-responsive layout using `layout="tabular"`. Flex and Grid layout wraps components in a DIV and grid layout
+ further wraps a row in a DIV.  These DIVs may now be styled by utilizing `row` and `column` components which optionally
+ represent the cell and may be utilized to attach `style`, `styleClass` and `id`.
+
+`layout="grid"` supports `column` and `row`.
+`layout="flex"` supports `column`.
+
+`row` may contain individual components, or additional `column` cells, within the grid responsive layout.
+The respective cell style will be utilized from the provided `columnClasses` and from any `styleClass` on the
+component.  If more components are contained within the row component then the respective `columnClasses` will
+be utilized based `columnClasses[column%columns]`.
+
+
+```xml
+<p:panelGrid 
+        columns="2" 
+        layout="grid"
+        columnClasses="ui-grid-col-8,ui-grid-col-4,ui-grid-col-8,ui-grid-col-4">
```

Comment:
You set columns to to but add 4 column-classes. Is this by intention?

Reply:
This conversation was resolved by a comment on the main discussion.


> I was considering showing an example of a row with more than 2 columns embedded in the row. These columns would then require a column class.
> 
> I might make a small change to the indexing internal to use all the column classes if necessary.
> 
> The current indexing is using index % columns but within a row i am thinking of using index % columnClasses.length.